core/src/main/java/brooklyn/enricher/basic/Transformer.java

```diff
@@ -102,13 +95,65 @@ public void setEntity(EntityLocal entity) {
         }
     }
 
+    /** returns a function for transformation, for immediate use only (not for caching, as it may change) */
+    @SuppressWarnings("unchecked")
+    protected Function<SensorEvent<T>, U> getTransformation() {
```

Comment:
Why might the result of `getTransformation()` change from one call to the other? Is it not based on the config values? If so and if we're catering for config values changing, then we could override `doReconfigureConfig(...)`.